src/subcommands/export_public_key.rs

```diff
@@ -39,3 +90,55 @@ impl ExportPublicKey {
         Ok(())
     }
 }
+
+fn curve_oid(curve: EccFamily, key_bits: usize) -> Result<ObjectIdentifier> {
+    let curve_oid = match curve {
+        // SEC random curves over prime fields.
+        EccFamily::SecpR1 => match key_bits {
+            192 => picky_asn1_x509::oids::secp192r1(),
+            224 => picky_asn1_x509::oids::secp224r1(),
+            256 => picky_asn1_x509::oids::secp256r1(),
+            384 => picky_asn1_x509::oids::secp384r1(),
+            521 => picky_asn1_x509::oids::secp521r1(),
+            _ => return print_error(curve, key_bits),
+        },
+        // SEC Koblitz curves over prime fields.
+        // OIDs are not defined in picky_asn1_x509::oids and in RFC5480.
+        // Use values from https://www.secg.org/sec2-v2.pdf#subsection.A.2
+        EccFamily::SecpK1 => match key_bits {
+            192 => ObjectIdentifier::try_from("1.3.132.0.31").unwrap(),
+            224 => ObjectIdentifier::try_from("1.3.132.0.32").unwrap(),
+            256 => ObjectIdentifier::try_from("1.3.132.0.10").unwrap(),
+            _ => return print_error(curve, key_bits),
+        },
+        // SEC Koblitz curves over binary fields
+        EccFamily::SectK1 => match key_bits {
+            233 => picky_asn1_x509::oids::sect233k1(),
+            283 => picky_asn1_x509::oids::sect283k1(),
+            409 => picky_asn1_x509::oids::sect409k1(),
+            571 => picky_asn1_x509::oids::sect571k1(),
+            _ => return print_error(curve, key_bits),
+        },
+        // SEC random curves over binary fields
+        EccFamily::SectR1 => match key_bits {
+            233 => picky_asn1_x509::oids::sect233r1(),
+            283 => picky_asn1_x509::oids::sect283r1(),
+            409 => picky_asn1_x509::oids::sect409r1(),
+            571 => picky_asn1_x509::oids::sect571r1(),
+            _ => return print_error(curve, key_bits),
+        },
+        _ => {
+            error!("Unsupported Ecc family \"{}\"", curve);
+            return Err(ToolErrorKind::NotSupported.into());
+        }
+    };
+    Ok(curve_oid)
+}
```

Comment:
This is really useful! I am wondering if it should be in the `psa-crypto` or `parsec-interface` crate to be shared by other parts (in the future, this does not have to be now)? We have something similar already [here](https://github.com/parallaxsecond/parsec/blob/87bf19a654dba0d5842ed2f1f69bb053b72ee166/src/providers/pkcs11/utils.rs#L127).